code/game/turfs/turf.dm

```diff
@@ -291,11 +291,17 @@
 	lighting_corner_SW = old_lighting_corner_SW
 	lighting_corner_NW = old_lighting_corner_NW
 
+	var/area/thisarea = get_area(W)
 	//static Update
 	if(SSlighting.initialized)
 		recalculate_directional_opacity()
 
-		W.static_lighting_object = old_lighting_object
+		if(thisarea.static_lighting)
+			W.static_lighting_object = old_lighting_object || new /datum/static_lighting_object(src)
+		else
+			W.static_lighting_object = null
+			if(old_lighting_object)
+				qdel(old_lighting_object)
```

Comment:
lighting objects only del on force del, also not a qdel null or does the lighting object handle that

Reply:
This is the old object, i.e. its only a var in this proc. The turf ref is nulled right above. This is basically for the situation where a spacetile (or some other nostaticlighting tile) replaces a tile with static lighting.

Updated with forced tho